src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -373,12 +373,24 @@ private static int GetMinRequiredLength(Regex r)
         [InlineData("\\w\\d123|\\w\\dabc", "\\w\\d(?:123|abc)")]
         // Auto-atomicity
         [InlineData("a*b", "(?>a*)b")]
-        [InlineData("a*b+", "(?>a*)b+")]
-        [InlineData("a*b{3,4}", "(?>a*)b{3,4}")]
+        [InlineData("a*b+", "(?>a*)(?>b+)")]
+        [InlineData("a*b*", "(?>a*)(?>b*)")]
+        [InlineData("a*b+c*", "(?>a*)(?>b+)(?>c*)")]
+        [InlineData("a*b*c*", "(?>a*)(?>b*)(?>c*)")]
+        [InlineData("a*b*c*|d*[ef]*", "(?>a*)(?>b*)(?>c*)|(?>d*)(?>[ef]*)")]
+        [InlineData("(a*)(b*)(c*)", "((?>a*))((?>b*))((?>c*))")]
+        [InlineData("a*b{3,4}", "(?>a*)(?>b{3,4})")]
+        [InlineData("[ab]*[^a]*", "[ab]*(?>[^a]*)")]
+        [InlineData("a??", "")]
```

Comment:
`[InlineData("(abc)??", "")` might be worth adding too? I don't see that higher up.

Reply:
Functionally with the source generator / compiler you'll end up with empty, but we don't actually get that today after parsing / reducing, due to the structure of the optimization that runs.  We end up with the equivalent of `(abc){0}?`, but the nature of how this test is verifying things means I can't actually write that.  Opportunities for another day...